### ISSUE TITLE
[air] xgboost/lightgbm trainer's validation result differ between online and offline

### ISSUE THREAD

**Issue by xwjiang2010** [ray-project/ray#34211]
See context [here](https://discuss.ray.io/t/ray-tune-metrics-not-consistent-with-offline-evaluation/10113/6)

I think the problem is that each worker will do evaluation using its own shard of data and we just randomly pick one and report. But I would like @Yard1 to confirm my understanding when he comes back.

**Comment by Yard1** [comment]
Let me just paste my reply from the thread:

> We only report the results from rank 0 worker, but each actor should have the same evaluation set - we should be only sharding the training data, not evaluation data. If we are not doing that, this is a bug - I will need to double check the logic in LightGBMTrainer. I am not sure whether the evaluation is done before or after the gradients are synced. If it is the latte and the data is the same on all workers, then all workers should return the same metrics (I think that is the case).

This will most likely need a change on the xgboost-ray side

**Comment by xwjiang2010** [comment]
https://github.com/ray-project/xgboost_ray/blob/master/xgboost_ray/main.py#L1062
this seems assigning different eval dataset to different actors?

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
